ouroboros-consensus-test/test-consensus/Test/Consensus/HardFork/Combinator.hs

```diff
@@ -108,6 +110,9 @@ instance Arbitrary TestSetup where
       testSetupEpochSize <- abM $ EpochSize <$> choose (1, 10)
       testSetupK         <- SecurityParam   <$> choose (2, 10)
       -- TODO why does k=1 cause the nodes to only forge in the first epoch?
+
+      -- TODO @js: think about what value should be used here.
+      testSetupS         <- GenesisWindowLength <$> choose (2, 10)
```

Comment:
Does this mean that the value of `s` is not derived from `f` and `k`?

Reply:
It is derived from `f` and `k` where there exist. This one specifically has been changed to `2k`.